src/conn/conn_cache_pool.c

```diff
@@ -117,7 +117,16 @@ __wt_cache_pool_config(WT_SESSION_IMPL *session, const char **cfg)
 		    "Attempting to join a cache pool that does not exist: %s",
 		    pool_name);
 
+	/*
+	 * At this point we have a cache pool to use. We need to take its
+	 * lock. We need to drop the process lock first to avoid deadlock
+	 * and acquire in the proper order.
+	 */
+	__wt_spin_unlock(session, &__wt_process.spinlock);
```

Comment:
I feel like releasing this lock here opens the possibility of the cache pool being destroyed by another thread. I think we'd need to validate the `cache pool` we'd saved is still OK after getting the process lock again.

Did you think about changing the other usage to acquire the process lock before the cache pool lock?

Reply:
The destroy cannot happen because the creation code does not set the `WT_CONN_CACHE_POOL` flag until the very end, after we've relocked in the proper order. If a destroy tries to happen in the interim when the create code is at the point of redoing the locks, the destroy is a no-op and just returns due to that flag.

I think that the locking and create/destroy could use a general review but I think this change solves this bug.